source/mir/random/algorithm.d

```diff
@@ -28,7 +28,7 @@ struct RandomField(G, D, T)
     Constructor.
     Stores the pointer to the `gen` engine.
     +/
-    this()(ref G gen, D var) { _gen = &gen; _var = var; }
+    this()(ref G gen, D var) { _gen = (() @trusted => &gen)(); _var = var; }
```

Comment:
This is not safe. `gen` can be provided is provided by a user.

Reply:
Oh, let me think more..

Reply:
@9il Maybe two constructors: a `@safe` one that accepts `G*`, and a `@system` one that accepts `ref G`. Then it's up to the caller to either obtain the pointer `@safe`ly or mark the code where he obtains the pointer `@trusted` if he wants his code to be `@safe`.